File: scala-gradle-build/common/src/main/scala/chobit/common/Strings.scala
package chobit.common

object Strings {

  /**
    * 判断字符串是否为空
    */
  def isBlank(source: String): Boolean = {
    if (null == source)
      return true
    val length = source.length
    if (0 == length)
      return true
    for (i <- 0 until length) {
      if (!Character.isWhitespace(source.charAt(i)))
        return false
    }
    true
  }


  /**
    * 判断字符串是否为空
    */
  def isNotBlank(source: String): Boolean = !isBlank(source)


  /**
    * 判断字符串是否全部为数字
    */
  def isNumeric(source: String): Boolean = {
    if (null == source)
      return false
    for (i <- 0 until source.length) {
      if (!Character.isDigit(source.charAt(i)))
        return false
    }
    true
  }


  /**
    * 判断字符串是否全部为数字
    */
  def isNotNumeric(source: String): Boolean = !isNumeric(source)

}

File: bytebuddy-explore/src/main/java/org/chobit/bytebuddy/tools/Strings.java
package org.chobit.bytebuddy.tools;

import java.util.UUID;


/**
 * 字符串操作工具类
 */
public abstract class Strings {


    /**
     * 生成UUID字符串
     */
    public static String uuid() {
        return UUID.randomUUID().toString();
    }


    /**
     * 判断字符串是否不为空
     */
    public static boolean isNotBlank(final CharSequence cs) {
        return !isBlank(cs);
    }


    /**
     * 判断字符串是否为空
     */
    public static boolean isBlank(final CharSequence cs) {
        int strLen;
        if (cs == null || (strLen = cs.length()) == 0) {
            return true;
        }
        for (int i = 0; i < strLen; i++) {
            if (!Character.isWhitespace(cs.charAt(i))) {
                return false;
            }
        }
        return true;
    }


    private Strings() {
        throw new UnsupportedOperationException("Private constructor, cannot be accessed.");
    }

}
